reconcile/utils/openshift_resource.py

```diff
@@ -87,21 +89,21 @@ def obj_intersect_equal(self, obj1, obj2):
                             self.kind, k
                         )
                     ]
-                    if diff or not self.obj_intersect_equal(obj1_v, obj2_v):
+                    if diff or not self.obj_intersect_equal(obj1_v, obj2_v, depth + 1):
```

Comment:
nit: It's a python best practice to use keys for kwargs
```suggestion
                    if diff or not self.obj_intersect_equal(obj1_v, obj2_v, depth=depth + 1):
```